### DISCUSSION TITLE
Is it possible to monitor inbound API calls to SF using Nebula Logger?

### DISCUSSION THREAD

**Discussion by victorng8** [jongpie/NebulaLogger#685] [Q&A] [answered]
Example :
If I call SF REST API `https://MyDomainName.my.salesforce.com/services/data/v60.0/sobjects/Account `from external system, how can we get the request and response on Salesforce as a record/file/etc ?

**Comment by jongpie** [comment, marked as answer]
Hi @victorng8 - as far as I'm aware, the only way to monitor Salesforce's standard REST API endpoints is to by [monitoring events like `ApiEvent` with Salesforce Shield/Event Monitoring](https://developer.salesforce.com/docs/atlas.en-us.platform_events.meta/platform_events/sforce_api_objects_apievent.htm). This requires paying for [Salesforce Shield](https://www.salesforce.com/products/shield/), or paying for the Salesforce Event Monitoring add-on subscription - since it's not functionality that all orgs can use, I have not integrated this into Nebula Logger.

However, one option that several orgs use is to [create custom Apex REST services](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_rest_code_sample_basic.htm), incorporate Nebula Logger into their services, and use those endpoints instead of Salesforce's standard endpoints. It's definitely more work with this approach, but provides greater control (including the ability to add logging).

Hope this helps! Let me know if you have any other questions.
